### DISCUSSION TITLE
Appropriate mesh file created in gmsh

### DISCUSSION THREAD

**Discussion by Ali1990dashti** [idaholab/moose#18106] [Q&A Meshing] [answered]
Dear community,
I am new in using moose and want to import my mesh into moose. I am using an open source code developed on top of moose (called [Tiger](https://github.com/MGK-Lab/tiger)) to change my mesh from .msh file into a exodus file to visualize it in paraview. I created my mesh using gmsh. Then I used a syntax of Tiger to change the mesh (`--mesh-only`) but the result which I am seeing in paraview is completely different from what I created in gmsh. My mesh has two blocks and 7 surfaces (uploaded fig). I want to do THM modelling using mentioned code (Tiger) on the uploaded mesh file but I can see the mesh I created. I have no idea what is the problem of my mesh. How can I check if my mesh is correctly prepared for moose? I do appreciate any help in advance.

![gmsh_result](https://user-images.githubusercontent.com/62764899/122374345-95afc080-cf62-11eb-82b6-857ed52085fb.jpg)

[mesh.zip](https://github.com/idaholab/moose/files/6669162/mesh.zip)

**Comment by ajacquey** [comment]
Hi Ali,

It would help if you share your `.geo` file you used to create the mesh with gmsh.
Without it, I can only give you some hints to check.

First, did you set the physical blocks in the `.geo` file?
It is usually done with:
```
Physical Volume(0) = {1};
Physical Volume(1) = {2};
```
with 1 and 2 being the volume you creates and 0 and 1 being the `block` number that you can use in the moose input file.

Then, you can do the same with the boundary surfaces:
```
Physical Surface("bottom") = {1};
```

Hope it helps.

**Reply by Ali1990dashti** [reply]
Dear @ajacquey, I do not use `.geo` files to create my mesh. I am using python API of gmsh and my syntax is in python. If you open my msh file with gmsh, volumes and surfaces look fine, but I have no idea why moose is not reading them. If you have look at my mesh, you can see that I have  defined both my 2d and 3d `Physical Names`.

**Reply by ajacquey** [reply]
Hi @Ali1990dashti ,
Yes, I can indeed see that the mesh has physical volumes and surfaces. The log message from moose reads fine when using the `--mesh-only` option.
From paraview, I can only see the bottom part of your model in a volume representation. Any other representation yields nothing.
![Screenshot from 2021-06-17 09-55-35](https://user-images.githubusercontent.com/7922526/122410762-35b11e00-cf52-11eb-8f4a-34f6a2fad94c.png)

Maybe there is something wrong in the top part of your model. I have never experienced this issue so I'm not sure I will be of any help.

**Reply by ajacquey** [reply]
My bad, I think it's the top part of your model (block unit_3). 
The issue might come from the basement block

**Reply by Ali1990dashti** [reply]
@ajacquey , Thanks for your help. Visualized section is the basement block. At least in my mesh it is the basement block. Another problem is that it does not read my surfaces. Strange thing is that I can convert it without any error but when I visualize it, it looks different.

**Reply by ajacquey** [reply]
I can also load and display the physical surfaces properly.
However, they all have really large values of ObjectId (around 7000-9000,  physical surface ID). 
Could you try to set these values to something closer to unity?

**Reply by ajacquey** [reply]
![Screenshot from 2021-06-17 10-07-58](https://user-images.githubusercontent.com/7922526/122412839-e8ce4700-cf53-11eb-8703-c1d1aa29a9ff.png)

This is what I mean with the ObjectId values.

**Reply by Ali1990dashti** [reply]
@ajacquey do you mean the tag number of surfaces and volumes should be reduced and be for example less than 100?

**Reply by ajacquey** [reply]
Yes, 
Your 2 subdomains have now 7406 (unit_3) and 9290 (basement) as IDs.
And your sidesets 9291-9296.

When I first opened the exodus file in Paraview, I got a warning about large numbers but I think I clicked on 'Ok, don't show again' so I cannot see this warning anymore.

I don't know how your python gmsh API works but can you try to set the IDs directly instead of giving names for your physical volumes and surfaces?

**Reply by ajacquey** [reply]
FYI, I have the same issue with peacock

**Reply by GiudGiud** [reply]
@ajacquey @Ali1990dashti any news on this?

**Comment by Ali1990dashti** [comment, marked as answer]
@ajacquey , I changed the mesh but still I cannot see any result in paraview. I installed the last version of moose and also use paraview 5.9. 

[Changed_mesh.zip](https://github.com/idaholab/moose/files/6670991/Changed_mesh.zip)

**Reply by Ali1990dashti** [reply]
@ajacquey , Can I ask which version of paraview and conda are you using? I am seeing different things (almost nothing!!!) in my paraview. For moose also I installed it just two days ago. It has some problems with anaconda and I used following lines to install moose:
```
conda create -n moose -q -y
conda activate moose
conda install moose-tools
conda install moose-libmesh
```

**Reply by ajacquey** [reply]
@Ali1990dashti 
I have Paraview 5.9.1
I updated moose and conda in May:
git commit 5cda0d307c on 2021-05-10

**Reply by Ali1990dashti** [reply]
@ajacquey, problem was my mesh. I was creating physical groups for blocks of my mesh but because of a mistake in a for loop, these physical groups were creating two times. Thanks for all your support. Sorry for my late feedback.
